gravitee-am-model/src/main/java/io/gravitee/am/model/UserId.java

```diff
@@ -17,6 +17,17 @@
 
 public record UserId(String id, String externalId, String source) {
 
+    public static UserId parse(String rawUserId) {
+        var parts = rawUserId.split(":");
+        return switch (parts.length) {
+            // internal id
+            case 1 -> new UserId(parts[0], null, null);
+            // `gis` claim format
+            case 2 -> new UserId(null, parts[1], parts[0]);
+            default -> throw new IllegalArgumentException("Malformed userId: " + rawUserId);
+        };
+    }
```

Comment:
I think the UserId should manage the logic of spliting the GIS claims as it is mainly only useful on the GW side not the mAPI one. In addition, using split is not good, we should use indexOf and use substring as we can't assume the externalId will not contains this separator.